### PR TITLE
Improve accessibility of dropdown select button

### PR DESCRIPTION
**Pull request by sergiou87**
xref. https://github.com/github/accessibility-audits/issues/4936

## Description

This PR adds the necessary aria attributes to improve the accessibility of the dropdown select button.

### Screenshots

https://github.com/desktop/desktop/assets/1083228/83c60c8d-42ef-43ac-82ab-6a78ec524b21

## Release notes

Notes: [Improved] Improve accessibility of dropdown select buttons

### CODE REVIEW

File: app/src/ui/dropdown-select-button.tsx [hunk 1]
```diff
@@ -121,25 +140,22 @@ export class DropdownSelectButton<
       return
     }
 
-    const foundCurrentIndex = [...buttons].findIndex(b =>
-      b.className.includes('focus')
-    )
+    const foundCurrentIndex =
+      document.activeElement === null
+        ? -1
+        : [...buttons].indexOf(document.activeElement)
```
Comment (sergiou87) [inline app/src/ui/dropdown-select-button.tsx:146]: Not sure if using `document.activeElement` is the best way, but for some reason the class `focus` is not applied anymore. I couldn't find if that was a Chrome thing (and therefore an Electron upgrade broke it) or if it was something custom we did.

In any case, using `document.activeElement` works and we also do a similar check [in the TextBox component](https://github.com/desktop/desktop/blob/4fd78b52bbca68aa291899329d64b4164754491d/app/src/ui/lib/text-box.tsx#L146)

**Review by tidy-dev — CHANGES_REQUESTED** [review]
✨ This definitely works a lot better. 

Whether the item is selected or not is not announced and that would seem to be because of: 
- aria-selected is always false since we are not setting the `focusedOptionIndex` in the state
- when tabbing the `focusedOptionIndex` is not set.

Additionally, I think the focus interaction for tabbing and mouse hover is not expected.

Other things.. when I created this with buttons.. I thought I was getting keyboard navigation for free. but.. nothing is life is free...

I believe we would be better suited by not using buttons, but building a traditional menu out of list items and managing the focus in the js.  The [W3C Action Menu Button Example](https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/examples/menu-button-actions/) provides [example js](https://www.w3.org/WAI/content-assets/wai-aria-practices/patterns/menu-button/examples/js/menu-button-actions.js) that shows how to handle the key press actions.

If you would like, I can take this on.. since it is my mess we are cleaning up. :)

File: app/src/ui/dropdown-select-button.tsx (hunk 1 above)
Comment (tidy-dev) [reply app/src/ui/dropdown-select-button.tsx:146]: Yeah.. I feel like I vaguely remember a few months back something about focus stuff changing. Maybe this was impacted then too.

**Comment by sergiou87** [comment]
Closing in favor of https://github.com/desktop/desktop/pull/17271

**Pull request closed without merging**